### ISSUE TITLE
Convert Tab's context flyout into a XAML control

### ISSUE THREAD

**Issue by leonMSFT** [microsoft/terminal#7841]
Tabs currently create their context flyout menu items in a big create function. It would be nice to convert that into a custom XAML control so that the more menu items we decide to add to the flyout, the easier it will be to manage. We should also design it in a way where different types of tabs (i.e. SettingsUI tab vs TerminalTab) can choose what menu items they would like to use.

**Comment by DHowett** [comment]
Funny enough, I tried to do this (just having the context menu in a .xaml file) and it was a total pain to hook up the event handlers without introducing another class just to implement the events that could come _out_ of the context menu :smile:

**Comment by leonMSFT** [comment]
_Damn_, it's not as easy as exposing the `MenuFlyoutItems` in the IDL and just being like `menuflyoutitem.Click{function}`?

**Comment by leonMSFT** [comment]
Oh nvm I see, you tried it out by only having a XAML file?

**Comment by DHowett** [comment]
Yep. Just a XAML file w/ a flyout, then one with a usercontrol w/ a flyout, then then then

**Comment by mdtauk** [comment]
Have you looked into using Xaml UserControls or CustomControls?

**Comment by DHowett** [comment]
@mdtauk most of our project is UserControls or CustomControls.

**Comment by mdtauk** [comment]
> @mdtauk most of our project is UserControls or CustomControls.

Fair enough then lol

**Comment by zadjii-msft** [comment]
What if we just passed a `ShortcutActionDispatch` into the (theoretical) `TabContextMenuControl`? Then we could have the items in the flyout dispatch actions to that, routing them up to the Page as usual. We'll probably want that anyways, esp if we want to let users customize the content of that flyout.
